program-runtime/src/cpi.rs

```diff
@@ -580,6 +602,25 @@ pub fn translate_accounts_rust<'a>(
         check_aligned,
     )?;
 
+    if invoke_context
```

Comment:
Any particular reason you chose this location rather than putting the metering inside `translate_account_infos` after the `check_account_infos` call? I think that would be a better place because then we meter CU's before doing any allocation.